backend/src/main/java/com/redhat/cloud/notifications/db/InternalRoleAccessResources.java

```diff
@@ -0,0 +1,51 @@
+package com.redhat.cloud.notifications.db;
+
+import com.redhat.cloud.notifications.models.InternalRoleAccess;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.transaction.Transactional;
+import java.util.Collection;
+import java.util.List;
+import java.util.UUID;
+
+@ApplicationScoped
+public class InternalRoleAccessResources {
+
+    @Inject
+    EntityManager entityManager;
+
+    public List<InternalRoleAccess> getByApplication(UUID applicationId) {
+        final String query = "FROM InternalRoleAccess WHERE application_id = :applicationId";
+        return entityManager.createQuery(query, InternalRoleAccess.class)
+                .setParameter("applicationId", applicationId)
+                .getResultList();
+    }
+
+    public List<InternalRoleAccess> getAll() {
+        final String query = "FROM InternalRoleAccess";
+        return entityManager.createQuery(query, InternalRoleAccess.class).getResultList();
+    }
+
+    @Transactional
+    public InternalRoleAccess addAccess(InternalRoleAccess access) {
+        entityManager.persist(access);
+        return access;
+    }
+
+    @Transactional
+    public void removeAccess(UUID internalRoleAccessId) {
+        final String query = "DELETE FROM InternalRoleAccess WHERE id = :id";
+        entityManager.createQuery(query)
+                .setParameter("id", internalRoleAccessId)
+                .executeUpdate();
+    }
+
+    public List<InternalRoleAccess> getByRole(Collection<String> role) {
```

Comment:
> role or roles?

It can be roles, yes.

> Why Collection? Can't we use something more specific?

Why should I use something more specific? It's passed to the database and I don't need to do any operation with that.